Tests/test_stdconsole.py

```diff
@@ -27,6 +27,28 @@
     class clr(object):
         IsDebug = False
 
+class IronPythonVariableContext(object):
+    def __init__(self, variable, value, sep=os.pathsep, prepend=False):
+        from System import Environment
+        self._variable = variable
+        self._value = value
+        self._prepend = prepend
+        self._sep = sep
+        self._oldval = Environment.GetEnvironmentVariable(self._variable)
+        
+    def __enter__(self):
+        from System import Environment
+        if self._prepend:
+            Environment.SetEnvironmentVariable(self._variable, "%s%s%s" % (self._value, self._sep, self._oldval))
+        else:
+            Environment.SetEnvironmentVariable(self._variable, self._value)
+
+    def __exit__(self, *args):
+        from System import Environment
+        if self._oldval:
```

Comment:
Why the check? Wouldn't you want to remove the environment variable if it was not defined before?